tests/framework/helpers/StringHelperTest.php

```diff
@@ -264,4 +264,39 @@ public function testWordCount()
         $this->assertEquals(1, StringHelper::countWords('крем-брюле'));
         $this->assertEquals(1, StringHelper::countWords(' слово '));
     }
+
+    /**
+     * @dataProvider base64UrlEncodedStringsProvider
+     * @param $input
+     * @param $base64UrlEncoded
+     */
+    public function testBase64UrlEncode($input, $base64UrlEncoded)
+    {
+        $encoded = StringHelper::base64UrlEncode($input);
+        $this->assertEquals($base64UrlEncoded, $encoded);
+
+        $safeUrlStringRegEx = sprintf('/[a-zA-Z0-9\-_%%]{%d}/', strlen($encoded));
+        $this->assertRegExp($safeUrlStringRegEx, $encoded);
```

Comment:
we already know that `$base64UrlEncoded` equals `$encoded`. so what does this additional assertion accomplish?

It seems to be testing the test vector, not the unit under test.